### ISSUE TITLE
Cannot enqueue Quit after invoking quit.

### ISSUE THREAD

**Issue by pcnetsoft-projects** [bradzacher/mysqldump#10]
I'm working to get backup for multi database in for loop, but i got this error.

`events.js:154
      throw er; // Unhandled 'error' event
      ^
Error: Cannot enqueue Quit after invoking quit.
    at Protocol._validateEnqueue (/home/mohammad/wallet_webservices/node_modules/mysql/lib/protocol/Protocol.js:202:16)
    at Protocol._enqueue (/home/mohammad/wallet_webservices/node_modules/mysql/lib/protocol/Protocol.js:135:13)
    at Protocol.quit (/home/mohammad/wallet_webservices/node_modules/mysql/lib/protocol/Protocol.js:88:23)
    at Connection.end (/home/mohammad/wallet_webservices/node_modules/mysql/lib/Connection.js:242:18)
    at /home/mohammad/wallet_webservices/node_modules/mysqldump/index.js:158:20
    at /home/mohammad/wallet_webservices/node_modules/async/lib/async.js:52:16
    at /home/mohammad/wallet_webservices/node_modules/async/lib/async.js:550:17
    at /home/mohammad/wallet_webservices/node_modules/async/lib/async.js:544:17
    at _arrayEach (/home/mohammad/wallet_webservices/node_modules/async/lib/async.js:85:13)
    at Immediate.taskComplete (/home/mohammad/wallet_webservices/node_modules/async/lib/async.js:543:13)

`

**Comment by webcaetano** [comment]
@pcnetsoft-projects i don't have time to work on a feature to fix this today.
But this is a problem because the last patch, If you use the version 1.1.0 It will probably work just fine. 

Thanks for the report, will looking on that soon.

**Comment by Akamaozu** [comment]
@pcnetsoft-projects check out if this works for you.

https://github.com/Akamaozu/mysqldump/tree/multi-use

**Comment by pcnetsoft-projects** [comment]
Did you solve the problem ?!

**Comment by Akamaozu** [comment]
@pcnetsoft-projects i wrote a small test case to show dumping in a loop works. 
Try the branch I linked ... it should work for you.

@webcaetano is busy so he can't fix this yet. I'm trying to make things easier for him by working on a solution he can simply review and add.

You can help by testing the solution I've written to see if it works for you. I'd really appreciate it.

**Comment by pcnetsoft-projects** [comment]
You have a problem that you used the same Connection for another calls, and it's already closed, the second thing is that you used this line `connection.end()`, and this function is canceled from Mysql Nodejs team.

**Comment by Akamaozu** [comment]
@pcnetsoft-projects you're right about using the same connection for multiple calls.

That's why the main part of my fix is using a new mysql connection every time you try to make a dump.

https://github.com/Akamaozu/mysqldump/blob/multi-use/index.js#L83-L90

About `connection.end`, this is straight from node-mysql's readme:

> There are two ways to end a connection. Terminating a connection gracefully is done by calling the end() method:
> 
> ```
> connection.end(function(err) {
>   // The connection is terminated now
> });
> ```
> 
> This will make sure all previously enqueued queries are still before sending a COM_QUIT packet to the MySQL server. If a fatal error occurs before the COM_QUIT packet can be sent, an err argument will be provided to the callback, but the connection will be terminated regardless of that.
> 
> **Source**: https://github.com/felixge/node-mysql#terminating-connections

**Comment by bradzacher** [comment]
fixed #44

**Comment by bmk10** [comment]
implmenting the end functioon keeps script running 

```
con.end(function(err) {
	console.log(err);
  // The connection is terminated now
});
```
